### PR TITLE
editable "needed date" on mediation pages

### LINKED ISSUE #530: Allow mediator to edit a request's needed on date
Allow mediator to edit the "needed on" date of a request. (What will happen to the sort order when they do this?)

request mediation - edit date

### PR DESCRIPTION
**Pull request by jmartin-sul**
~~- UI problems~~
  ~~- arrows for paging calendar don't show up~~
  ~~- even if using text box as per original design, there doesn't seem to be a setting that auto-updates the display text on edit success~~
  ~~- the `Done` button by the `(+)` row expansion button gets squished and has a line break in it~~

~~also there are no tests.~~

~~the edit itself works fine though, and the options for date formatting do work nicely.  if you select a date and refresh the page, you can see that it gets updated (or did for me, in testing on my laptop).~~

~~annotated the WIP code, comments explaining what's broken and what works use extra comment chars so that they stick out a bit visually, hopefully without setting hound off.~~

sul-requests-530_upd-needed-on-date

closes #530

### CODE REVIEW

**Comment by jmartin-sul** [comment]
plenty of other stuff for hound to complain about, apparently

File: app/assets/stylesheets/modules/admin.scss [hunk 1]
```diff
@@ -36,9 +36,24 @@
...
 
 .dashboard-table {
   text-align: center;
 }
+
+.editable-date {
+  .icon-arrow-left {
+    @extend .glyphicon;
+    @extend .glyphicon-arrow-left;
+  }
```
Comment (jmartin-sul) [inline app/assets/stylesheets/modules/admin.scss]: is the `.icon-arrow-left` block what hound is complaining about?  seems like the line below is empty, so not sure what to do here...

**Comment by jmartin-sul** [comment]
the calendar UI for this should now work correctly, and this PR should be ready for review.  arrows for month selection are visible and functional, the text in the row is updated when a new date is selected, and the `Done` button renders correctly.  and there's now a test for the new feature.

thanks to @jkeck for help in ironing out datepicker UI issues and debugging tests.

File: app/assets/javascripts/application.js [hunk 2]
```diff
@@ -13,10 +13,19 @@
...
+//= require bootstrap-editable-rails
 //= require list.min.js
 //= require no_js
 //= require trunk8
 //= require_tree .
+
+$(document).on('turbolinks:load', function() {
+  $('.editable').editable();
+  $("[data-toggle='tooltip']").tooltip();
+});
```
Comment (jkeck) [inline app/assets/javascripts/application.js]: Can we add this into a separate file?

File: app/views/admin/_library_table.html.erb [hunk 3]
```diff
@@ -23,7 +23,24 @@
             <%= t('.all_approved') %>
           </span>
         </td>
-        <td class='needed_date'><%= time_tag(request.needed_date, :quick) %></td>
+        <td class='needed_date'>
+          <span data-toggle='tooltip' data-placement='left' title="edit 'Needed on' date">
+            <%=
+              # ## somewhat following the example here: https://github.com/sul-dlss/library_hours_rails/blob/master/app/views/libraries/_location_hours.html.erb#L25
+              # ## that uses a link, but it doesn't actually seem to matter what the element is, editable takes it over and handles styling, and divs and spans seem
+              # ## to have the same behavior, other than the icon on hover
```
Comment (jkeck) [inline app/views/admin/_library_table.html.erb]: Is this comment still valid?

Comment (jmartin-sul) [inline app/views/admin/_library_table.html.erb]: nope, will remove

**Comment by jmartin-sul** [comment]
* erroneous comment deleted
* tried to move the JS @jkeck pointed out to its own file.  but that broke things and neither of us could figure out why, so i just left a comment explaining why it's directly in `application.js` (neither of us thought it was worth further debugging time, though if someone else has an idea of how to fix it...)
* commits squashed

ready for re-review?  build appears to almost be passing, except for a couple not-so-intermittent failures on the travis run that's yet to complete successfully... but i expect that will go green after another retry or two.

File: app/assets/javascripts/application.js [hunk 4]
```diff
@@ -13,10 +13,22 @@
...
 //= require no_js
 //= require trunk8
 //= require_tree .
+
+// normally, we'd put this in its own file, but when we tried
+// that, the functionality broke, and it didn't seem worth spending
+// the time to debug further.
+$(document).on('turbolinks:load', function() {
+  $('.editable').editable();
+  $("[data-toggle='tooltip']").tooltip();
```
Comment (jmartin-sul) [inline app/assets/javascripts/application.js]: i like this more than escaped quotes (or maybe i'm misunderstanding the complaint)

**Pull request merged**